apps/condo/domains/meter/components/Meters/CreateMeterForm.tsx

```diff
@@ -0,0 +1,198 @@
+        const numberOfTariffs = values.numberOfTariffs || 1
+        
+        if (!initialRecord) {
+            createMeterAction({
+                ...values,
+                numberOfTariffs,
+                organization: { connect: { id: organizationId } },
+                ...getCommonMeterFields(values),
+            })
+        } else {
+            updateMeterAction({
+                ...values,
+                ...getCommonMeterFields(values),
+            }, { id: initialRecord.id })
+        }
+        
+        //TODO: notification about saving meter and reset form
+    },
+    [createMeterAction, getCommonMeterFields, initialRecord, organizationId, updateMeterAction])
+
+    const connectedPropertyIitialValues = useMemo(() => !initialRecord ? ({}) : ({
+        propertyId: get(initialRecord, 'property.id'),
+        unitName: isPropertyMeter ? undefined : get(initialRecord, 'unitName'),
+        unitType: isPropertyMeter ? undefined : get(initialRecord, 'unitType'),
+    }), [initialRecord, isPropertyMeter])
+
+
+    const ErrorToFormFieldMsgMapping = {
+        [EXISTING_METER_NUMBER_IN_SAME_ORGANIZATION]: {
+            name: 'number',
+            errors: [MeterWithSameNumberIsExistMessage],
+        },
+        [EXISTING_METER_ACCOUNT_NUMBER_IN_OTHER_UNIT]: {
+            name: 'accountNumber',
+            errors: [AccountNumberIsExistInOtherUnitMessage],
+        },
+    }
+
+    const getHandleSelectPropertyAddress = useCallback((form) => (_, option) => {
+        setSelectedPropertyId(String(option.key))
+    }, [])
+
+    const handleDeselectPropertyAddress = useCallback(() =>  {
+        setSelectedPropertyId(null)
+    }, [])
+
+    const handleCancelEditing = useCallback(() =>  {
+        router.push(`/meter/${meterType === METER_TAB_TYPES.propertyMeter ? 'propertyDevice' : 'device'}/${initialRecord.id}`)
+    }, [initialRecord, meterType, router])
+
+
+    return (
+        <PageWrapper style={WRAPPER_STYLE}>
+            <FormWithAction>
+                {
+                    ({ form }) => (
+                        <AddressAndUnitInfo 
+                            form={form}
+                            getHandleSelectPropertyAddress={getHandleSelectPropertyAddress}
+                            handleDeselectPropertyAddress={handleDeselectPropertyAddress}
+                            isMatchSelectedProperty={isMatchSelectedProperty}
+                            meterType={meterType}
+                            organizationId={organizationId}
+                            selectedPropertyId={selectedPropertyId}
+                            property={property}
+                            propertyLoading={propertyLoading}
+                            setSelectedUnitName={setSelectedUnitName}
+                            initialValues={connectedPropertyIitialValues}
+                        />
+                    )
+                }
+            </FormWithAction>
+            <FormWithAction
```

Comment:
removed two FormWithAction, left one
and renamed BaseMeterForm -> BaseMeterFormFields (since it's just fields and the form itself is in  CreateMeterForm or UpdateMeterForm components)